_pages/documentation/gem5-stdlib/5-local-resources-support.md

```diff
@@ -0,0 +1,161 @@
+---
+layout: documentation
+title: Local Resources Support in gem5
+parent: gem5-standard-library
+doc: gem5 documentation
+permalink: /documentation/gem5-stdlib/local-resources-support
+author: Kunal Pai, Harshil Patel
+---
+
+# Adding a Workload to gem5
+
+This tutorial will walk you through the process of creating a WorkloadResource in gem5 and testing it, through the new gem5 Resources infrastructure introduced in gem5 v23.0.
+
+A workload is set to a board in gem5 through the following line:
+
+``` python
+board.obtain_resource(<ID_OF_WORKLOAD>)
+```
+
+The WorkloadResource with ID '<ID_OF_WORKLOAD>' will be parsed and it will be used to construct the function call it defines. The function call specified in the `"function"` field of the Workload JSON is then executed on the board, along with any parameters it has defined in the `"additional_parameters"` field.
+
+## Introduction
+
+The gem5 Resources infrastructure allows adding a local JSON data source that can be added to the main gem5 Resources MongoDB database. We will use the local JSON data source to add a new WorkloadResource to gem5.
+
+## Prerequisites
+
+This tutorial assumes that you already have a pre-compiled Resource that you want to make into a WorkloadResource.
+
+## Defining the Workload
+
+### Defining the Resource JSON
+
+The first step is to define the Resource that is used in a WorkloadResource. In case the Resource already exists in gem5, you may skip this step.
+
+Let's assume that the Resource we want to wrap in a WorkloadResource is compiled for `RISC-V`, categorized as a `binary`, and has the name `binary-resource`. We can define this Resource in a JSON object as follows:
```

Comment:
The name `binary-resource` here is confusing. I would choose something like `my-benchmark` or `foo-benchmark`